lib/components/tag/tag.dart

```diff
@@ -0,0 +1,213 @@
+import 'dart:html';
+import 'package:katikati_ui_lib/components/button/button.dart';
+import 'package:katikati_ui_lib/components/logger.dart';
+
+var logger = Logger('tag.dart');
+
+enum TagStyle {
+  None,
+  Green,
+  Yellow,
+  Red,
+  Important,
+}
+
+class TagView {
+  SpanElement renderElement;
+  SpanElement _tagText;
+  SpanElement _tagActions;
+
+  String _text;
+  String _tagId;
+
+  bool _selectable;
+  bool _editable;
+  bool _editableOnAdd;
+  bool _removable;
+  bool _acceptable;
+  bool _suggested;
+
+  TagStyle _tagStyle;
+
+  ButtonElement _editButton;
+  ButtonElement _deleteButton;
+  ButtonElement _confirmButton;
+  ButtonElement _cancelButton;
+  ButtonElement _acceptButton;
+
+  // some default callbacks, so we can use some behaviours like reset without implementing the raw onclicks
+  void Function(String) onEdit = (_) {};
+  void Function() onDelete = () {};
+  void Function() onSelect = () {};
+  void Function() onAccept = () {};
+
+  TagView(this._text, this._tagId,
+      {String groupId,
+      bool selectable = true,
+      bool editable = false,
+      bool editableOnAdd = false,
+      bool removable = false,
+      bool acceptable = false,
+      bool suggested = false,
+      TagStyle tagStyle = TagStyle.None,
+      bool disableDoubleClickEdit = false}) {
```

Comment:
can we swap these two around so we have a consecutive series of bools?

I would also be inclined to suggest we write `disableDoubleClickEdit` in the affirmative (e.g. `doubleClickToEdit`), and set it to `false` by default. What do you think?